docs/app.md

```diff
@@ -190,6 +191,31 @@ myApp.isRunning();
 myApp.start();
 ```
 
+### App `restart`
+
+This method saves the current state of the app before stopping it.
+It then starts the app again with the preserved state attributes.
+
+```js
+var myApp = new Marionette.Toolkit.App();
+
+var initialState = {
+  foo: 'bar'
+};
+
+myApp.start({
+  state: initialState
+});
+
+myApp.setState('foo', 'baz');
+
+// Nothing is logged
```

Comment:
similar to the request on the test this code example doesn't explain much about what can be expected.  I think having an `onStart` or and `onBeforeStart` log what the state is would be more useful.